test/Serialization/autolinking.swift

```diff
@@ -22,6 +22,11 @@
 // RUN: %target-swift-frontend -runtime-compatibility-version none -emit-ir -parse-stdlib -module-name someModule -module-link-name module %S/../Inputs/empty.swift -autolink-force-load | %FileCheck --check-prefix=FORCE-LOAD %s
 // RUN: %target-swift-frontend -runtime-compatibility-version none -emit-ir -parse-stdlib -module-name someModule -module-link-name 0module %S/../Inputs/empty.swift -autolink-force-load | %FileCheck --check-prefix=FORCE-LOAD-HEX %s
 
+// RUN: %target-swift-frontend -emit-module -parse-stdlib -o %t -module-name someModule -module-link-name module %S/../Inputs/empty.swift -public-autolink-library anotherLib
+// RUN: %target-swift-frontend -disable-autolinking-runtime-compatibility-dynamic-replacements -runtime-compatibility-version none -emit-ir -lmagic %s -I %t > %t/public-autolink.txt
+// RUN: %FileCheck %s < %t/public-autolink.txt
+// RUN: %FileCheck -check-prefix=PUBLIC-DEP %s < %t/public-autolink.txt
+
 // Linux uses a different autolinking mechanism, based on
 // swift-autolink-extract. This file tests the Darwin mechanism.
 // UNSUPPORTED: autolink-extract
```

Comment:
If I'm reading this right, we aren't actually running this test on Linux, the platform where we need this change to work. Is it being properly tested? Is there a different test we could use to verify that it's working on non-Darwin platforms?

Reply:
Good catch. In theory, this new option is not limited to Linux platform, but should be tested on Linux also. I added a test case to verify it.